### PR TITLE
codestyle: Add missing type annotations to some files in ima directory and add them to mypy

### PR DESCRIPTION
**Pull request by stefanberger**

### CODE REVIEW

**Comment by stefanberger** [comment]
Known missing dependencies:

```
keylime/ima/ima.py:531: error: Call to untyped function "verify_signature" in typed context  [no-untyped-call]
keylime/ima/ima.py:590: error: Call to untyped function "verify_signature" in typed context  [no-untyped-call]
```

File: keylime/ima/ima.py [hunk 1]
```diff
@@ -584,8 +584,8 @@ def unbundle_ima_policy(allowlist_bundle, verify: bool = True) -> Dict[str, Any]
                 message="Verifier is enforcing signature validation, but no public key was provided.", code=405
             )
 
-        allowlist_sig = base64.b64decode(allowlist_bundle.get("sig"))
-        allowlist_sig_key = base64.b64decode(allowlist_bundle.get("key"))
+        allowlist_sig = base64.b64decode(cast(bytes, allowlist_bundle.get("sig")))
+        allowlist_sig_key = base64.b64decode(cast(bytes, allowlist_bundle.get("key")))
 
         if not signing.verify_signature(allowlist_sig_key, allowlist_sig, allowlist_raw):
```
Comment (aplanas) [inline keylime/ima/ima.py:590]: seems that `verify_signature` from `keylime/signing.py` is missing the return type

File: keylime/ima/file_signatures.py [hunk 2]
```diff
@@ -145,10 +145,10 @@ def get_pubkey_by_keyidv2(self, keyidv2: int) -> Optional[SupportedKeyTypes]:
         """Get a public key object given its keyidv2"""
         return self.ringv2.get(keyidv2)
 
-    def to_json(self) -> Dict[str, List[Union[int, str]]]:
+    def to_json(self) -> Dict[str, Union[List[int], List[str]]]:
         """Convert the ImaKeyring into a JSON object"""
         fmt = serialization.PublicFormat.SubjectPublicKeyInfo
-        obj = {}
+        obj = cast(Dict[str, Union[List[int], List[str]]], {})
```
Comment (aplanas) [inline keylime/ima/file_signatures.py:151]: Maybe `obj: Dict[str, Union[List[int], List[str]]] = {}`

```python
# python3
Python 3.10.8 (main, Oct 19 2022, 07:46:20) [GCC] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import Dict, List, Union
>>> obj: Dict[str, Union[List[int], List[str]]] = {}
>>> __annotations__
{'obj': typing.Dict[str, typing.Union[typing.List[int], typing.List[str]]]}
```

File: keylime/ima/ima.py (hunk 1 above)
Comment (stefanberger) [reply keylime/ima/ima.py:590]: I just added type-annotations for signing.py to the other PR.

File: keylime/ima/file_signatures.py (hunk 2 above)
Comment (stefanberger) [reply keylime/ima/file_signatures.py]: Fixed. Thanks.

**Review by aplanas — APPROVED** [review]

**Review by THS-on — APPROVED** [review]

**Pull request merged**